src/App.tsx

```diff
@@ -1,27 +1,94 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+import { ButtonsNames } from './types/ButtonsNames';
+
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [lastButton, setLastButton] = useState(ButtonsNames.NONE);
+
+  const handleGoodsLoading = useCallback(
+    (
+      getGoods: () => Promise<Good[]>,
+      buttonName: ButtonsNames,
+    ) => {
+      if (buttonName === lastButton) {
```

Comment:
```suggestion
      if (buttonName === lastButton && !hasError) {
```

User needs a possibility to refetch in case of error

Reply:
i implement it in catch method, if i have error i set last button none